src/main/java/org/jfree/chart/axis/TickUnits.java

```diff
@@ -126,6 +128,7 @@ public TickUnit get(int pos) {
      * @return A tick unit that is larger than the supplied unit.
      */
     @Override
+    @SuppressWarnings("index") // binary search on list

```

Comment:
Shouldn't binary search indices (`@SearchIndexFor`) handle this code?  If not, please make the warning suppression more specific.  The same comment applies to line 154 and maybe elsewhere.